apps/ledger-live-mobile/src/components/ModalLock.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { setModalLock } from "../actions/appstate";
+
+/**
+ * All steps of a DeviceAction that have an ongoing operation on
+ * the device should disable the closing of the modal. If we are
+ * transfering something (installing an app, a language, CLS, etc)
+ * we don't want to break the flow.
+ */
+const LockModal = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(setModalLock(true));
+    return () => {
+      dispatch(setModalLock(false));
+    };
+  }, [dispatch]);
+
+  return null;
+};
+
+export default LockModal;
```

Comment:
Should it be memoized to avoid re-rendering it when it's not necessary ? This will reduce the number of dispatch occurring

Reply:
I tested it (as per the video) and it didn't seem to cause any problem. Worst case scenario is we unlock and lock again.
I think we can live with it

Reply:
Ok !